Odata-docs/webapi-8/fundamentals/query-options.md

```diff
@@ -0,0 +1,547 @@
+    "value": [
+        {
+            "Name": "Customer 1"
+        },
+        {
+            "Name": "Customer 2"
+        },
+        {
+            "Name": "Customer 3"
+        }
+    ]
+}
+```
+
+We can return `Customers` with their related `Orders`. In this case we use `$expand`.
+
+```http
+GET http://localhost:6285/odata/Customers?$expand=Orders
+```
+
+```json
+{
+    "@odata.context": "http://localhost:6285/odata/$metadata#Customers(Orders())",
+    "value": [
+        {
+            "Id": 1,
+            "Name": "Customer 1",
+            "Age": 31,
+            "Orders": [
+                {
+                    "Id": 1002,
+                    "Price": 35,
+                    "Quantity": 2
+                },
+                {
+                    "Id": 1001,
+                    "Price": 10,
+                    "Quantity": 10
+                }
+            ]
+        },
+        {
+            "Id": 2,
+            "Name": "Customer 2",
+            "Age": 32,
+            "Orders": [
+                {
+                    "Id": 1004,
+                    "Price": 20,
+                    "Quantity": 20
+                },
+                {
+                    "Id": 1003,
+                    "Price": 70,
+                    "Quantity": 5
+                }
+            ]
+        },
+        {
+            "Id": 3,
+            "Name": "Customer 3",
+            "Age": 33,
+            "Orders": [
+                {
+                    "Id": 1006,
+                    "Price": 15,
+                    "Quantity": 50
+                },
+                {
+                    "Id": 1005,
+                    "Price": 40,
+                    "Quantity": 15
+                }
+            ]
+        }
+    ]
+}
+```
+
+We can combine various query options.
+In the request below:
+- Select only the `Name` property from `Customer` entity.
+- Expand the related `Orders`.
+- Filter the `Orders` and return only the ones whose `Id` is greater than `1004`.
+- Order the `Customers` by `Name` in descending order.
+
+```http
+http://localhost:6285/odata/Customers?$select=Name&$expand=Orders($filter=Id gt 1004)&$orderby=Name desc
+```
+
+```json
+{
+    "@odata.context": "http://localhost:6285/odata/$metadata#Customers(Name,Orders())",
+    "value": [
+        {
+            "Name": "Customer 3",
+            "Orders": [
+                {
+                    "Id": 1006,
+                    "Price": 15,
+                    "Quantity": 50
+                },
+                {
+                    "Id": 1005,
+                    "Price": 40,
+                    "Quantity": 15
+                }
+            ]
+        },
+        {
+            "Name": "Customer 2",
+            "Orders": []
+        },
+        {
+            "Name": "Customer 1",
+            "Orders": []
+        }
+    ]
+}
+```
+
+### Limiting Query Options
+We can limit the type of query option that can be used while calling a certain API action.
```

Comment:
@KenitoInc a common source of confusion from customers query options is how AllowedQueryOptions interacts with config options. e.g. if a user doesn't explicitly specify options=>.filter() can I activate with AllowedQueryOptions.Filter

An example issue here https://github.com/OData/AspNetCoreOData/issues/667